liboptv/src/track.c

```diff
@@ -0,0 +1,1409 @@
+ * region is only in the allowed directions.
+ * Arguments:
+ * vec3d point position in physical space
+ * track_par *tpar set of tracking parameters
+ * control_par *cpar set of control parameters for the num_cams
+ * Calibration *cal calibration per camera to find a projection of a 3D vertex
+ * of a cuboid in the image space.
+ * Returns the arrays xr,xl,yd,yu (right, left, down, up) per camera
+ * for the search of a quader (cuboid), given in pixel distances, relative to the 
+ * point of search.
+ */
+
+void searchquader(vec3d point, double xr[4], double xl[4], double yd[4], \
+                  double yu[4], track_par *tpar, control_par *cpar, Calibration **cal){
+    int i, pt, dim;
+    vec3d mins, maxes;
+    vec2d corner, center;
+    vec3d quader[8];
+
+
+
+    vec_set(mins, tpar->dvxmin, tpar->dvymin, tpar->dvzmin);
+    vec_set(maxes, tpar->dvxmax, tpar->dvymax, tpar->dvzmax);
+        
+    /* 3D positions of search volume - eight corners of a box */
+    for (pt = 0; pt < 8; pt++) {
+        vec_copy(quader[pt], point);
+        for (dim = 0; dim < 3; dim++) {
+            if (pt & 1<<dim) {
+                quader[pt][dim] += maxes[dim];
+            } else {
+                quader[pt][dim] += mins[dim];
+            }
+        }
+    }
+
+
+    /* calculation of search area in each camera */
+    for (i = 0; i < cpar->num_cams; i++) {
+        
+        /* initially large or small values */
+        xr[i] = 0;
+        xl[i] = cpar->imx;
+        yd[i] = 0;
+        yu[i] = cpar->imy;
+        
+        /* pixel position of a search center */
+        point_to_pixel (center, point, cal[i], cpar);
+        
+        /* mark 4 corners of the search region in pixels */
+        for (pt = 0; pt < 8; pt++) {
+            point_to_pixel (corner, quader[pt], cal[i], cpar);
+
+            if (corner[0] < xl[i] ) xl[i] = corner[0];
+            if (corner[1] < yu[i] ) yu[i] = corner[1];
+            if (corner[0] > xr[i] ) xr[i] = corner[0];
+            if (corner[1] > yd[i] ) yd[i] = corner[1];
+        }
+        
+        if (xl[i] < 0 ) xl[i] = 0;
+        if (yu[i] < 0 ) yu[i] = 0;
+        if (xr[i] > cpar->imx)
+            xr[i] = cpar->imx;
+        if (yd[i] > cpar->imy)
+            yd[i] = cpar->imy;
+
+        /* eventually xr,xl,yd,yu are pixel distances relative to the point */
+        xr[i] = xr[i]     - center[0];
+        xl[i] = center[0] - xl[i];
+        yd[i] = yd[i]     - center[1];
+        yu[i] = center[1] - yu[i];
+    }
+}
+
+/* sortwhatfound sourts the list of candidates in foundpix array
+ * Arguments:
+ * foundpix array, item[] which is as long as: num_cam*MAX_CANDS candidates
+ * integer * counter - pointer to the list of the candidates
+ * integer num_cams - number of cameras in the experiment (typically 1-4)
+ * Returns the sorted array foundpix item[] and a pointer *counter to
+ * an integer number of different candidates
+ */
+
+void sortwhatfound (foundpix item[], int *counter, int num_cams)
```

Comment:
Naming is awful. What kind of things were found? "what" or "stuff" are not good words. same goes to the header comments and to comments inside. This is an opportunity to actually explain what's going on to the next poor soul who needs to modify this code.